driver-core/src/main/com/mongodb/connection/ServerDescription.java

```diff
@@ -773,85 +773,111 @@ public boolean equals(final Object o) {
 
         ServerDescription that = (ServerDescription) o;
 
-        if (maxDocumentSize != that.maxDocumentSize) {
+        if (shouldPublishChangeEvent(this, that)) {
```

Comment:
Though I see why you did this to DRY up the code, let's just leave the equals method alone.

Reply:
I have reverted the equals method.